HPCC-15840 Wrong side serializer used when no helper key serializer

jakesmith opened the pull request:
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

richardkchapman commented:
@ghalliday Please review

HPCCSmoketest commented:
Automated Smoketest 
Sha: 371a4b3ec550ede9caeb851826196d8da1283b35
Build: success
Install hpccsystems-platform-community_6.0.2-rc3.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

jakesmith committed:
HPCC-15840 Master not passing aux rowIf in join.

If no keyserializer was generated, then the standard row
serializer was used. The slaves correctly setup based on the
primary partitioning side, the master however didn't pass the
primary row interface and based on the other side.
And hence caused serialization problems.

NB: This was only an issue if no keySerializer generated in a
global join.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest 
Sha: 75df55aeb5d59349b03b4ca81171eece994c70bf
Build: success
Install hpccsystems-platform-community_6.0.2-rc3.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK
